### DISCUSSION TITLE
I am missing an easy way to control volume level

### DISCUSSION THREAD

**Discussion by bachmarc** [alandtse/alexa_media_player#2765] [General]
I have seen that the the media player provides devices and assigned entities to HA.
The only thing I really miss is a simple way to control the volume level.

I know there is a service (which i failed to call from appdaemon due to nested unknown parameters) but even when I use it in yaml automations it feels like a tumor attached oddly to the other components.
I call the service to set it but cannot read the original volume level to return to it after I pushed the notification.

I would like to find an entity that controls the volume of a device in the integration. Or a parameter to the notify service...

[a comment by an automated account is not shown]

**Comment by bachmarc** [comment]
so be it ...

**Comment by danielbrunt57** [comment]
@bachmarc 
Rather than creating a "bug" report, you could have created a "feature request" or more appropriately, opened a "new discussion".  You could also post a question over in the HA Community Forum.  I'll move this to discussions for you..

**Comment by danielbrunt57** [comment]
I've just submitted a new PR #2766 which will add a new attribute to the AMP media players: `previous_volume`.
This should make life a _little_ easier!

![image](https://github.com/user-attachments/assets/8b93aefa-42c4-4681-b724-c16e82cd7b1d)

![image](https://github.com/user-attachments/assets/9b08df5e-c08e-4a3a-9cfb-ecd1e0af4e52)

**Comment by bachmarc** [comment]
THX!
